src/Features/Homescreen.php

```diff
@@ -74,18 +75,63 @@ public function add_user_data_fields( $user_data_fields ) {
 	 * Registers home page.
 	 */
 	public function register_page() {
+		// Register a top-level item for users who cannot view the core WooCommerce menu.
+		if ( ! $this->is_admin_user() ) {
+			add_filter( 'woocommerce_prevent_admin_access', '__return_false' );
```

Comment:
This will need to be replaced, but was a short-term fix.  Core allows the following capabilities without redirection:

* `edit_posts`
* `manage_woocommerce`
* `view_admin_dashboard`

The `view_admin_dashboard` is not a traditional capability and was created for this purpose, however it's also not a default available capability and may create confusion for store owners.

Alternatively (and more realistically) we may need to capture all the registered pages in wc-admin and check against the user capabilities to see if any match before turning this off.  Will update accordingly if we decide to go the direction of this PR.

Reply:
I take it that we only need to register this for the redirection? Given we remove it from the menu.

Also what makes this menu item show up in the `global $menu` when the one registered below doesn't show up, is it just the use of `order`?

Reply:
>I take it that we only need to register this for the redirection? Given we remove it from the menu.

The `woocommerce_prevent_admin_access` filter is for redirection, but this menu item is registered to allow users to access `wc-admin` pages otherwise they may not have the capability to access a `wc-admin` page even though the page has a different capability.

WordPress only cares about the `?page=wc-admin` portion and ignores `path`.

>Also what makes this menu item show up in the global $menu when the one registered below doesn't show up, is it just the use of order?

The one below has a parent of `woocommerce` which requires certain capabilities the user may or may not have.  This item is registered as a top-level item.

Reply:
Gotcha, thanks for the clarification @joshuatf, my WP PHP knowledge is a bit limited, this helps a lot.